src/vm/stackwalk.cpp

```diff
@@ -1778,7 +1778,7 @@ StackWalkAction StackFrameIterator::Filter(void)
                                 pCurrTracker = pCurrTracker->GetPreviousExceptionTracker();
                             }
 
-                            if (pCurrTracker != NULL)
+                            if ((pCurrTracker != NULL) && !pCurrTracker->IsEHClauseFilter())
```

Comment:
@sdmaclea I can repro the GC hole on amd64 too with GCStress 2. Let me get my head around the exact way it happens. It seems it would make sense to remove this commit from this PR and create a separate PR for it so that we can merge in the remainder of the commits without unnecessary delay.